src/jit/lower.cpp

```diff
@@ -628,20 +620,14 @@ GenTree* Lowering::LowerSwitch(GenTree* node)
 
                 // Now, build the conditional statement for the current case that is
                 // being evaluated:
-                // GT_JTRUE
-                //   |__ GT_COND
-                //          |____GT_EQ
-                //                 |____ (switchIndex) (The temp variable)
-                //                 |____ (ICon)        (The actual case constant)
-                GenTreePtr gtCaseCond =
-                    comp->gtNewOperNode(GT_EQ, TYP_INT, comp->gtNewLclvNode(tempLclNum, tempLclType),
-                                        comp->gtNewIconNode(i, TYP_INT));
-                /* Increment the lvRefCnt and lvRefCntWtd for temp */
+                // GT_CMP switchValue, caseValue
+                // GT_JCC(EQ)
```

Comment:
I was thinking more that it mapped better to how the ISA deals with it.  One compare that does all the flags, and branches that encode what bits to branch on.  This allows for single compare multiple consumer kind of things rather than keeping a single compare with the particular boolean relation in it.  (i.e. LT)